wallet/src/ui/app/components/network-switch/NetworkSelector.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useMemo, useCallback } from 'react';
+
+import { API_ENV_TO_INFO, API_ENV } from '_app/ApiProvider';
+import BsIcon from '_components/bs-icon';
+import { useAppSelector, useAppDispatch } from '_hooks';
+import {
+    setApiEnv,
+    changeRPCNetwork,
+    setNetworkSelector,
+} from '_redux/slices/app';
+import { getTransactionsByAddress } from '_redux/slices/txresults';
+import store from '_store';
+
+import st from './Network.module.scss';
+
+const NetworkSelector = () => {
+    const selectedApiEnv = useAppSelector(({ app }) => app.apiEnv);
+    const dispatch = useAppDispatch();
+    const networkList: any = API_ENV_TO_INFO;
+    const netWorks = useMemo(
+        () =>
+            Object.keys(networkList).map((itm: string) => ({
+                style: { color: networkList[itm].color },
+                ...networkList[itm],
+                networkName: itm,
+            })),
+        [networkList]
+    );
+
+    const changeNetwork = useCallback(
+        (networkName: string) => () => {
+            const apiEnv = API_ENV[networkName as keyof typeof API_ENV];
+            dispatch(setNetworkSelector(true));
+            store.dispatch(setApiEnv(apiEnv));
+            dispatch(changeRPCNetwork()).unwrap();
+            dispatch(getTransactionsByAddress()).unwrap();
+        },
+        [dispatch]
+    );
+
+    return (
+        <div className={st['network-options']}>
+            <div className={st['network-header']}>RPC NETWORK</div>
+            <ul className={st['network-lists']}>
+                {netWorks.map((apiEnv, index) => (
+                    <li
+                        className={st['network-item']}
+                        key={`networkid-${index}`}
+                        onClick={changeNetwork(apiEnv.networkName)}
```

Comment:
In general I think is better to use the data fields of the node and avoid creating new functions every time.
```
<li
...
data-network={networkName}
>
```
and access it from the currentTarget of the event